src/test/scala/akka/persistence/cassandra/journal/FailingRecoverSpec.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.persistence.cassandra.journal
+/*
+ * Copyright (C) 2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+import akka.actor.Status.Failure
+import akka.persistence.cassandra.testkit.CassandraLauncher
+
+import scala.concurrent.duration._
+import akka.actor.{ ActorRef, ActorSystem, Props, _ }
+import akka.persistence.{ PersistentActor, _ }
+import akka.persistence.cassandra.CassandraLifecycle
+import akka.testkit.{ ImplicitSender, TestKit, TestProbe, _ }
+import com.typesafe.config.ConfigFactory
+import org.scalatest._
+
+import scala.concurrent.Await
+
+object FailingRecoverSpec {
+  val config = ConfigFactory.parseString(
+    s"""
+       |cassandra-journal.port = ${CassandraLauncher.randomPort}
+       |cassandra-snapshot-store.port = ${CassandraLauncher.randomPort}
+       |cassandra-journal.keyspace=StartupLoadSpec
+       |cassandra-snapshot-store.keyspace=StartupLoadSpecSnapshot
+    """.stripMargin
+  ).withFallback(CassandraLifecycle.config)
+
+  val failingConfig = ConfigFactory.parseString(
+    """
+      |cassandra-journal.connection.read-timeout=10ms
+      |cassandra-snapshot-store.connection.read-timeout=10ms
+    """.stripMargin
+  ).withFallback(config)
+
+  class ProcessorA(val persistenceId: String, receiver: ActorRef) extends PersistentActor {
+    def receiveRecover: Receive = {
+      case _ =>
+    }
+
+    def receiveCommand: Receive = {
+      case payload: String =>
+        persist(payload)(handle)
+    }
+
+    def handle: Receive = {
+      case payload: String =>
+        receiver ! payload
+        receiver ! lastSequenceNr
+        saveSnapshot(payload)
+    }
+
+    override protected def onRecoveryFailure(cause: Throwable, event: Option[Any]): Unit = {
+      receiver ! Failure(cause)
```

Comment:
since you want to test that it got stuck I think you should send another custom message to the receiver here, such as `Started`, and the same message should be sent when receiving `RecoveryCompleted` in `receiveRecover`. In that way you can test for only the got stuck and don't care about if some actually got through within the 10.millis